api/users/models.py

```diff
@@ -235,12 +235,18 @@ def get_organisation_join_date(self, organisation):
     def get_user_organisation(
         self, organisation: typing.Union["Organisation", int]
     ) -> UserOrganisation:
+        organisation_id = getattr(organisation, "id", organisation)
+
         try:
-            return self.userorganisation_set.get(organisation=organisation)
-        except UserOrganisation.DoesNotExist:
+            return next(
+                filter(
+                    lambda uo: uo.organisation_id == organisation_id,
+                    self.userorganisation_set.all(),
+                )
+            )
```

Comment:
Why can't a normal Django `.get` work here?

Reply:
This is because of the way that `prefetch_related` works. If we do a `get()` here, it triggers another query to the database because django isn't smart enough to realise it already has the data.

Reply:
That's good to know. Is this common enough of a pattern or should we put a small comment explaining the logic?

Reply:
It's (thankfully) not very common. I'll add a comment.